keras_cv/models/object_detection/retina_net/retina_net_inference_test.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+
+import pytest
+import tensorflow as tf
+
+import keras_cv
+
+
+class RetinaNetTest(tf.test.TestCase):
+    @pytest.fixture(autouse=True)
+    def cleanup_global_session(self):
+        # Code before yield runs before the test
+        yield
+        tf.keras.backend.clear_session()
+
+    def test_weight_setting(self):
+        x, y = _create_bounding_box_dataset(bounding_box_format="xywh")
+        retina_net, new_retina_net = _create_retina_nets(x, y, epochs=1)
+
+        new_retina_net.set_weights(retina_net.get_weights())
+
+        # check if all weights that show up via `get_weights()` are loaded
+        for retina_net_weight, post_load_weight in zip(
+            retina_net.get_weights(), new_retina_net.get_weights()
+        ):
+            self.assertAllEqual(retina_net_weight, post_load_weight)
+
+        for layer_original, layer_new in zip(
+            # manually check layers to make sure nothing is missed
+            _get_retina_net_layers(retina_net),
+            _get_retina_net_layers(new_retina_net),
+        ):
+            for weight, weight_new in zip(
+                layer_original.get_weights(), layer_new.get_weights()
+            ):
+                self.assertAllEqual(weight, weight_new)
+
+    @pytest.mark.skipif(os.name == "nt", reason="tempfile does not work on windows")
+    def test_weight_loading(self):
+        x, y = _create_bounding_box_dataset(bounding_box_format="xywh")
+        retina_net, new_retina_net = _create_retina_nets(x, y, epochs=1)
+
+        tmp = tempfile.mkdtemp()
+        retina_net.save_weights(f"{tmp}/checkpoint.h5")
```

Comment:
Can you also verify the tf format, which is probably the major use case?

https://www.tensorflow.org/api_docs/python/tf/keras/Model#load_weights

Reply:
done